sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -371,52 +579,64 @@ private static RequestContent GetStreamingEnabledRequestContent(RequestContent o
             return augmentedContent;
         }
 
-        internal HttpMessage CreateGetCompletionsRequest(string deploymentId, RequestContent content, RequestContext context)
+        internal RequestUriBuilder GetUri(string deploymentOrModelName, string operationPath)
         {
-            var message = _pipeline.CreateMessage(context, ResponseClassifier200);
-            var request = message.Request;
-            request.Method = RequestMethod.Post;
             var uri = new RawRequestUriBuilder();
             uri.Reset(_endpoint);
-            if (!string.IsNullOrEmpty(deploymentId))
+            if (string.IsNullOrEmpty(PublicOpenAIApiKey))
             {
                 uri.AppendRaw("/openai", false);
                 uri.AppendPath("/deployments/", false);
-                uri.AppendPath(deploymentId, true);
-                uri.AppendPath("/completions", false);
+                uri.AppendPath(deploymentOrModelName, true);
+                uri.AppendPath($"/{operationPath}", false);
                 uri.AppendQuery("api-version", _apiVersion, true);
             }
             else
             {
-                uri.AppendPath("/completions", false);
+                uri.AppendPath($"/{operationPath}", false);
             }
-            request.Uri = uri;
+            return uri;
+        }
+
+        internal HttpMessage CreateGetCompletionsRequest(
+            string deploymentOrModelName,
+            RequestContent content,
+            RequestContext context)
+        {
```

Comment:
These three methods can be reduced to 
"return CreatePostRequest(deploymentOrModelName, content, context, operationPath);"